src/goto-symex/symex_function_call.cpp

```diff
@@ -431,28 +427,20 @@ void goto_symext::return_assignment(statet &state)
 
   target.location(state.guard.as_expr(), state.source);
 
-  if(code.operands().size()==1)
-  {
-    exprt value=code.op0();
+  PRECONDITION(code.operands().size() == 1);
+  
+  exprt value=code.op0();
 
-    if(frame.return_value.is_not_nil())
-    {
-      code_assignt assignment(frame.return_value, value);
+  if(frame.return_value.is_not_nil())
+  {
+    code_assignt assignment(frame.return_value, value);
 
-      if(!base_type_eq(assignment.lhs().type(),
-                       assignment.rhs().type(), ns))
-        throw
-          "goto_symext::return_assignment type mismatch at "+
-          instruction.source_location.as_string()+":\n"+
-          "assignment.lhs().type():\n"+assignment.lhs().type().pretty()+"\n"+
-          "assignment.rhs().type():\n"+assignment.rhs().type().pretty();
+    INVARIANT(
+    	base_type_eq(assignment.lhs().type(), assignment.rhs().type(), ns),
+      "goto_symext::return_assignment type mismatch");
 
-      symex_assign(state, assignment);
-    }
-  }
-  else
-  {
-    if(frame.return_value.is_not_nil())
-      throw "return with unexpected value";
+    symex_assign(state, assignment);
   }
+
+  POSTCONDITION(!frame.return_value.is_not_nil());
```

Comment:
Replaced the `throw "return with unexpected value"` with just a `POSTCONDITION`. I think it's cleaner this way, and more self evident what it does, while not altering the semantics of the code.